saga/src/worker.ts

```diff
@@ -0,0 +1,81 @@
+import { InjectedSinks, NativeConnection, Worker, WorkerOptions } from '@temporalio/worker';
+import * as fs from 'fs';
+import { createActivities } from './activities';
+import { createClients } from './clients';
+import { Env, getEnv, isRemoteEnv } from './env';
+import { LoggerSinks } from './workflows';
+
+// worker
+async function run(env: Env) {
+  const sinks: InjectedSinks<LoggerSinks> = {
+    logger: {
+      info: {
+        fn(workflowInfo, message) {
+          console.log('workflow: ', workflowInfo.runId, 'message: ', message);
+        },
+        callDuringReplay: false, // The default
+      },
+      err: {
+        fn(workflowInfo, message) {
+          console.error('workflow: ', workflowInfo.runId, 'message: ', message);
+        },
+        callDuringReplay: false, // The default
+      },
+    },
+  };
+
+  // registrations
+  const singletonClients = await createClients();
+  const activities = createActivities(singletonClients) as any;
+
+  const opts: WorkerOptions = {
+    workflowsPath: require.resolve('./workflows'),
+    activities,
+    taskQueue: env.taskQueue,
+    sinks,
+  };
+
+  if (isRemoteEnv(env)) {
```

Comment:
This sample is already pretty big, I'd consider removing the complex setup from this.